packages/ui/src/forms/buildStateFromSchema/addFieldStatePromise.ts

```diff
@@ -516,15 +517,42 @@ export const addFieldStatePromise = async (args: AddFieldStatePromiseArgs): Prom
       state,
     })
   } else if (field.type === 'tabs') {
-    const promises = field.tabs.map((tab) => {
+    const hasTabWithCondition = field.tabs.some((tab) => tab.admin?.condition)
+    if (hasTabWithCondition && !field.id) {
+      throw new Error('Cannot use tab conditions without a tabs field ID')
```

Comment:
This is a good feature! Just a quick comment for now:
If we need `id` requirement, we probably want to move this check directly to the fields sanitize function:
https://github.com/payloadcms/payload/blob/beta/packages/payload/src/fields/config/sanitize.ts#L267-L273
you can check for both `admin.condition` there and `id`, so user will get this error as soon as they defined a tab with condition and without `id`. Does it make sense?

Reply:
Good suggestion! Have implemented this in https://github.com/payloadcms/payload/pull/8406/commits/d561d79e4f1a7d179ccbc9bca5b290e475295ff4.